iaso/api/stocks/serializers.py

```diff
@@ -0,0 +1,397 @@
+            "id",
+            "name",
+            "short_name",
+            "projects",
+            "org_unit_types",
+            "forms",
+            "children",
+            "display_units",
+            "display_precision",
+            "created_at",
+            "created_by",
+            "updated_at",
+            "updated_by",
+            "deleted_at",
+        ]
+        read_only_fields = ["created_at", "updated_at", "created_by", "updated_by", "deleted_at"]
+
+
+class StockKeepingUnitWriteSerializer(serializers.ModelSerializer):
+    projects = PrimaryKeyRelatedField(
+        many=True,
+        queryset=Project.objects.all(),
+        required=False,
+    )
+    org_unit_types = PrimaryKeyRelatedField(
+        many=True,
+        queryset=OrgUnitType.objects.all(),
+        required=False,
+    )
+    forms = PrimaryKeyRelatedField(
+        many=True,
+        queryset=Form.objects.all(),
+        required=False,
+    )
+
+    class Meta:
+        model = StockKeepingUnit
+        fields = [
+            "id",
+            "name",
+            "short_name",
+            "projects",
+            "org_unit_types",
+            "forms",
+            "display_units",
+            "display_precision",
+        ]
+
+    def validate(self, validated_data):
+        request = self.context.get("request")
+        if "projects" in validated_data and not all(
+            p.account == request.user.iaso_profile.account for p in validated_data["projects"]
+        ):
+            raise serializers.ValidationError("User doesn't have access to this project")
+
+        return validated_data
+
+    def update(self, instance, validated_data):
+        request = self.context.get("request")
+        instance.updated_by = request.user
+        return super().update(instance, validated_data)
+
+
+class StockItemSerializer(serializers.ModelSerializer):
+    org_unit = OrgUnitNestedSerializer()
+    sku = StockKeepingUnitNestedSerializer()
+    created_at = TimestampField()
+    updated_at = TimestampField()
+
+    class Meta:
+        model = StockItem
+        fields = [
+            "id",
+            "org_unit",
+            "sku",
+            "value",
+            "created_at",
+            "updated_at",
+        ]
+        read_only_fields = ["created_at", "updated_at"]
+
+
+class StockItemWriteSerializer(serializers.ModelSerializer):
+    sku = PrimaryKeyRelatedField(
+        queryset=StockKeepingUnit.objects.all(),
+        required=True,
+    )
+    org_unit = PrimaryKeyRelatedField(
+        queryset=OrgUnit.objects.all(),
+        required=True,
+    )
+    value = serializers.IntegerField(min_value=0)
+
+    class Meta:
+        model = StockItem
+        fields = [
+            "sku",
+            "org_unit",
+            "value",
+        ]
+
+    def validate(self, validated_data):
+        request = self.context.get("request")
+        sku = validated_data.get("sku")
+        if sku.account != request.user.iaso_profile.account:
```

Comment:
This will do the exact same thing, but might avoid fetching the relation
```suggestion
        if sku.account_id != request.user.iaso_profile.account_id:
```